devito/ops/transformer.py

```diff
@@ -30,8 +31,23 @@ def opsit(trees, count):
     expressions = []
     ops_expressions = []
 
+    it_range = []
     for tree in trees:
         expressions.extend(FindNodes(Expression).visit(tree.inner))
+        if isinstance(tree, IterationTree):
+            it_range = [it.bounds() for it in tree]
+
+    par_loop_array = Array(
+        name=namespace['ops_par_loop_range_name'],
+        dimensions=(DefaultDimension(name='len', default_value=sum(len(x)
+                                                                   for x in it_range)),),
+        dtype=np.int32,
+    )
+
+    par_loop_range = Expression(ClusterizedEq(Eq(
+        Symbol(par_loop_array.name),
```

Comment:
This is wrong. It won't generate the correct LHS.
It has to be the Array object, not a symbol of the same name

Reply:
The range is generated correctly. But I think I understand what you mean. It seems valid to have an Array in the espressions tree.
Just replace _Symbol(par_loop_array.name)_ to _par_loop_array_ keep working and solves this.

Reply:
> probably a few tests would be useful

I agree

Reply:
The range is, the initialisation isn't. `int range = { x_m, x_M, y_m, y_M}` is wrong. It should be `int range[4] = { x_m, x_M, y_m, y_M}`

Reply:
This is the return: _int par_loop_range[4] = {x_m, x_M, y_m, y_M};_
The value is being set when _the par_loop_array_ is created. See https://github.com/maelso/devito/blob/8c6ecbc27aaf15c88e0ae2b216213219fc31633e/devito/ops/transformer.py#L42

Reply:
Yup it's correct with your latest commit!